data/trie/patriciaMerkleTrie.go

```diff
@@ -400,7 +428,45 @@ func (tr *patriciaMerkleTrie) ResetOldHashes() [][]byte {
 	return oldHashes
 }
 
-// AppendToOldHashes appends the given hashes to the trie's oldHashes variable
-func (tr *patriciaMerkleTrie) AppendToOldHashes(hashes [][]byte) {
-	tr.oldHashes = append(tr.oldHashes, hashes...)
+// Snapshot creates a new database in which the current state of the trie is saved.
+// If the maximum number of snapshots has been reached, the oldest snapshot is removed.
+func (tr *patriciaMerkleTrie) Snapshot() error {
+	tr.mutOperation.Lock()
+	defer tr.mutOperation.Unlock()
+
+	err := tr.root.commit(false, 0, tr.db, tr.marshalizer, tr.hasher)
+	if err != nil {
+		return err
+	}
+
+	db, err := storageUnit.NewDB(
```

Comment:
these files could be precreated and you could do complete delete and commit on them.

Reply:
You mean another component that creates the database, and each time the snapshot function is called, a new database should be passed as a parameter?

Reply:
As you do not want to have infinite amount of snapshots, the snapshot files can be predetermined. So you can create N files, and write in those like in a sliding window. So you precreate N storageUnits - you could have a container for it, then get one out, change the snapshot there and that's it. You might not need a separate component, but a container with all the storageUnits for the tries might be sent in the constructor and not the config.